### PR TITLE
SearchBox nullref fix

### PR DESCRIPTION
**Pull request by dzearing**
Adjusted the searchbox styling to be valid. 

Also adjusted mergeStyles to be more resilient to invalid styling.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.styles.tsx [hunk 1]
```diff
@@ -33,12 +33,9 @@ export function getStyles(props: ISearchBoxStyleProps): ISearchBoxStyles {
...
           ':hover': {
             borderColor: palette.neutralDark,
-            $label: {
-              color: palette.black,
-              $iconContainer: {
-                color: palette.themeDark
-              }
-            }
+          },
+          ':hover $iconContainer': {
```
Comment (lijunle) [inline packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.styles.tsx]: How about the `$label` color?

File: packages/merge-styles/src/transforms/rtlifyRules.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export function rtlifyRules(
     const name = rulePairs[index] as string;
     const value = rulePairs[index + 1] as string;
 
-    if (value.indexOf(NO_FLIP) >= 0) {
+    if (typeof value.indexOf === 'string' && value.indexOf(NO_FLIP) >= 0) {
       rulePairs[index + 1] = value.replace(/\s*(?:\/\*\s*)?\@noflip\b(?:\s*\*\/)?\s*?/g, '');
     } else if (name.indexOf('left') >= 0) {
```
Comment (lijunle) [inline packages/merge-styles/src/transforms/rtlifyRules.ts]: Do we need to consider other calls?

File: packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.styles.tsx (hunk 1 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.styles.tsx]: It's not used. There is no label area.

File: packages/merge-styles/src/transforms/rtlifyRules.ts (hunk 2 above)
Comment (dzearing) [reply packages/merge-styles/src/transforms/rtlifyRules.ts]: Potentially, let me check.

Comment (dzearing) [reply packages/merge-styles/src/transforms/rtlifyRules.ts]: I have added a test to cover the bases. I think we're good.

**Comment by dzearing** [comment]
Hmmm. Not sure what is going on with this test.

**Comment by manishgarg1** [comment]
> export function rtlifyRules(

None of this code has any documentation about what it does.

---
Refers to: packages/merge-styles/src/transforms/rtlifyRules.ts:30 in 9ea2494. [](commit_id = 9ea2494dccb09709f7ae9b75e4286c75ac9b460e, deletion_comment = False)

**Comment by manishgarg1** [comment]
>       rulePairs[index] = name.replace('right', 'left');

should these be consts?

---
Refers to: packages/merge-styles/src/transforms/rtlifyRules.ts:48 in 9ea2494. [](commit_id = 9ea2494dccb09709f7ae9b75e4286c75ac9b460e, deletion_comment = False)

**Pull request merged**